tessel/src/lib.rs

```diff
@@ -82,86 +78,146 @@ pub struct PortGroup {
 /// use tessel::Port;
 /// ```
 pub struct Port {
-    // Path of the domain socket.
     socket: Arc<Mutex<PortSocket>>,
-    pins: HashMap<usize, Mutex<()>>,
 }
 
-pub struct Pin<'a> {
-    index: usize,
-    _guard: MutexGuard<'a, ()>,
-    socket: Arc<Mutex<PortSocket>>,
-}
+impl Port {
+    pub fn new(path: &str) -> Port {
+        // Create and return the port struct
+        Port {
+            socket: Arc::new(Mutex::new(PortSocket::new(path))),
```

Comment:
I think it's a little strange that the socket is thread-safe but the `ports` Box is designed to encourage developers to use it in a single-threaded manner.